mne/io/eeglab/eeglab.py

```diff
@@ -0,0 +1,140 @@
+                               unpack=True)
+    sph_phi, sph_theta = _topo_to_sph(angle, radius)
+    sph_radius = np.ones((eeg.nbchan, ))
+    x, y, z = _sph_to_cart(sph_theta / 180 * np.pi,
+                           sph_phi / 180 * np.pi, sph_radius)
+    if eog is None:
+        eog = [idx for idx, ch in enumerate(ch_names) if ch.startswith('EOG')]
+
+    for idx, ch in enumerate(info['chs']):
+        ch['loc'][:3] = np.array([x[idx], y[idx], z[idx]])
+        ch['unit'] = FIFF.FIFF_UNIT_V
+        ch['coord_frame'] = FIFF.FIFFV_COORD_HEAD,
+        ch['coil_type'] = FIFF.FIFFV_COIL_EEG,
+        ch['kind'] = FIFF.FIFFV_EEG_CH
+
+        if ch['ch_name'] in eog:
+            ch['coil_type'] = FIFF.FIFFV_COIL_NONE
+            ch['kind'] = FIFF.FIFFV_EOG_CH
+
+    return info
+
+
+def read_raw_set(fname, ch_fname, eog=None, verbose=None):
+    """Read an EEGLAB .set file
+
+    Parameters
+    ----------
+    fname : str
+        Path to the .set file.
+    ch_fname : str
+        Path to the file containing channel locations. Currently,
+        only .loc extension is supported.
+    eog : list or tuple
+        Names of channels or list of indices that should be designated
+        EOG channels. If None (default), the channel names beginning with
+        ``EOG`` are used.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of RawSet
+        A Raw object containing EEGLAB .set data.
+    """
+    return RawSet(fname, ch_fname, eog, verbose)
+
+
+class RawSet(_BaseRaw):
+    """Raw object from EEGLAB .set file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the .set file.
+    ch_fname : str
+        Path to the file containing channel locations. Currently,
+        only .loc extension is supported.
+    eog : list or tuple
+        Names of channels or list of indices that should be designated
+        EOG channels. If None (default), the channel names beginning with
+        ``EOG`` are used.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of RawSet
+        A Raw object containing EEGLAB .set data.
+    """
+    def __init__(self, fname, ch_fname, eog=None, verbose=None):
+        """Read EEGLAB .set file.
+        """
+        scaling = 1e-6
+        basedir = op.dirname(fname)
+        eeg = io.loadmat(fname, struct_as_record=False, squeeze_me=True)['EEG']
+
+        # read the data
+        data_fname = op.join(basedir, eeg.data)
+        logger.info('Reading %s' % data_fname)
+        data_fid = open(data_fname)
+        data = np.fromfile(data_fid, dtype=np.float64) * scaling
+        data = data.reshape((-1, eeg.nbchan)).T
+
+        # get info
+        if ch_fname is not None:
+            ch_fname = op.join(basedir, ch_fname)
+        info = _get_info(eeg, eog, ch_fname)
```

Comment:
Many set files already contain channel information. You can check `eeg.chanlocs` to see if it is true.
`eeg.chanlocs` is a matlab structure with following fields: `labels`, `Y`, `X`, `Z`, `sph_theta`, `sph_phi`, `sph_radius`, `theta`, `radius`, `type`, `ref`, `urchan`. If channel positions are present - all coordinate fields should be filled. `labels` contains channel names.
For example in matlab:
```matlab
EEG.chanlocs(5)
```
gives:
```
ans = 

        labels: 'E5'
             Y: -3.0769
             X: 10.1424
             Z: 1.2423
     sph_theta: -16.8763
       sph_phi: 6.6854
    sph_radius: 10.6714
         theta: 16.8763
        radius: 0.4629
          type: ''
           ref: 'average'
        urchan: 5
```